RFCs/FS-1084-The-UnknownEnum-pattern.md

```diff
@@ -0,0 +1,127 @@
+# F# RFC FS-1084 - The `UnknownEnum` pattern
+
+<!--The design suggestion [The `UnknownEnum` pattern](https://github.com/fsharp/fslang-suggestions/issues/822) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+-->
+
+- [ ] Approved in principle
+- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/822)
+- [x] [RFC Discussion](https://github.com/fsharp/fslang-design/issues/446)
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/FILL-ME-IN)
+
+# Summary
+
+A new active pattern should be added to FSharp.Core, named `UnknownEnum`, to match against undefined enum values.
+
+# Motivation
+
+Currently, for `match` expressions on enumerations (enums), the warning FS0104 which warns against unknown enum cases not being matched exists.
+
+However, to silence this warning, one has to either use a wildacard pattern, which has the downside of preventing the warning FS0025 from being emitted when a new enum case is added to the enum type, or disable this warning entirely, which has the downside of a `MatchFailureException` being raised once an invalid value is matched against, potentially as a result of deserialization which the developer cannot control.
+
+An `UnknownEnum` pattern would be an ideal solution to this, as it matches all cases that FS0104 would warn against, while keeping FS0025 warnings when a new case is added to the enum type. As a result, using `match` expressions on enums can be both safe, as in matching all potential values, and future-proof, as in keeping FS0025 warnings when a new case is added.
+
+# Detailed design
+
+A new active pattern, named `UnknownEnum` will be added to FSharp.Core:
+```fs
+type private UnknownEnumLookup<'Enum>() =
+    static member val Values = typeof<'Enum>.GetEnumValues() :?> 'Enum[]
+let (|UnknownEnum|_|) (enum:'Enum when 'Enum : enum<'Underlying>) =
+    if Array.IndexOf(UnknownEnumLookup<'Enum>.Values, enum) < 0 then
```

Comment:
This implementation is based on the **runtime** set of enum values.  Shouldn't it be based on the static set?

Reply:
Updated the document